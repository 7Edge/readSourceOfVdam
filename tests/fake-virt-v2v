#!/usr/bin/env python
# Copyright 2015 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
#
# Refer to the README and COPYING files for full details of the license
#

import sys
import argparse

parser = argparse.ArgumentParser()

parser.add_argument('args')
parser.add_argument('-ic', dest='inputConnection',
                    help='Set input libvirt connectio url')
parser.add_argument('-o', dest='output',
                    help='Set output method')
parser.add_argument('-of', dest='outputFormat',
                    help='Set output image format')
parser.add_argument('-oa', dest='outputAllocation',
                    help='Set output allocation format')
parser.add_argument('--vdsm-image-uuid', dest='vdsmImageId',
                    help='Vdsm image UUID', action='append')
parser.add_argument('--vdsm-vol-uuid', dest='vdsmVolId',
                    help='Vdsm volume UUID', action='append')
parser.add_argument('--vdsm-vm-uuid', dest='vdsmVmId',
                    help='VM UUID')
parser.add_argument('--password-file', dest='passwordFile',
                    help='Read connection password from a file')
parser.add_argument('--vdsm-ovf-output', dest='vdsmOvfOutput',
                    help='Output directory for ovf output')
parser.add_argument('-os', dest='outputStorage',
                    help='Output directory for the images')
parser.add_argument('--machine-readable', dest='machineReadable',
                    action='store_true',
                    help='Set the terminal output to be readable')
parser.add_argument('vmname')

options = parser.parse_args(sys.argv)
elapsed_time = 0


def write_output(msg):
    sys.stdout.write(msg)
    sys.stdout.flush()


def write_progress():
    global elapsed_time
    for i in range(101):
        write_output('    (%s/100%%)\r' % str(i))
        elapsed_time = elapsed_time + 1

write_output('[   %d.0] Opening the source -i libvirt\n' % elapsed_time)
elapsed_time = elapsed_time + 1
write_output('[   %d.0] Creating an overlay to protect\n' % elapsed_time)
elapsed_time = elapsed_time + 1

for i, o in enumerate(options.vdsmImageId):
    write_output('[  %d.0] Copying disk %d/2 to %s/%s/images/%s\n' %
                 (elapsed_time, i+1, options.outputStorage,
                  options.vdsmVmId, o))
    write_progress()
    write_output('[ %d.0] Creating output metadata\n' % elapsed_time)
    write_output('[ %d.0] Finishing off\n' % elapsed_time)
