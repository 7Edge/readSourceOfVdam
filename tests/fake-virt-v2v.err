libguestfs: trace: internal_autosync = 0
libguestfs: sending SIGTERM to process 12345
libguestfs: trace: shutdown = 0
libguestfs: trace: close
libguestfs: closing guestfs handle 0x1e265f0 (state 0)
