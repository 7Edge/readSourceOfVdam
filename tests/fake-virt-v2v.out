[   0.0] Opening the source -i libvirt
[   1.0] Creating an overlay to protect
[  2.0] Copying disk 1/2 to /rhev/data-center/00000000-0000-0000-0000-000000000006/00000000-0000-0000-0000-000000000007/00000000-0000-0000-0000-000000000005/images/00000000-0000-0000-0000-000000000001
target_file = 00000000-0000-0000-0000-000000000002
target_format = raw
target_estimated_size = 123456789
target_overlay = /var/tmp/v2vovl344e53.qcow2
    (0/100%)    (1/100%)    (2/100%)    (3/100%)    (4/100%)    (5/100%)    (6/100%)    (7/100%)    (8/100%)    (9/100%)    (10/100%)    (11/100%)    (12/100%)    (13/100%)    (14/100%)    (15/100%)    (16/100%)    (17/100%)    (18/100%)    (19/100%)    (20/100%)    (21/100%)    (22/100%)    (23/100%)    (24/100%)    (25/100%)    (26/100%)    (27/100%)    (28/100%)    (29/100%)    (30/100%)    (31/100%)    (32/100%)    (33/100%)    (34/100%)    (35/100%)    (36/100%)    (37/100%)    (38/100%)    (39/100%)    (40/100%)    (41/100%)    (42/100%)    (43/100%)    (44/100%)    (45/100%)    (46/100%)    (47/100%)    (48/100%)    (49/100%)    (50/100%)    (51/100%)    (52/100%)    (53/100%)    (54/100%)    (55/100%)    (56/100%)    (57/100%)    (58/100%)    (59/100%)    (60/100%)    (61/100%)    (62/100%)    (63/100%)    (64/100%)    (65/100%)    (66/100%)    (67/100%)    (68/100%)    (69/100%)    (70/100%)    (71/100%)    (72/100%)    (73/100%)    (74/100%)    (75/100%)    (76/100%)    (77/100%)    (78/100%)    (79/100%)    (80/100%)    (81/100%)    (82/100%)    (83/100%)    (84/100%)    (85/100%)    (86/100%)    (87/100%)    (88/100%)    (89/100%)    (90/100%)    (91/100%)    (92/100%)    (93/100%)    (94/100%)    (95/100%)    (96/100%)    (97/100%)    (98/100%)    (99/100%)    (100/100%)[ 103.0] Creating output metadata
[ 103.0] Finishing off
[  103.0] Copying disk 2/2 to /rhev/data-center/00000000-0000-0000-0000-000000000006/00000000-0000-0000-0000-000000000007/00000000-0000-0000-0000-000000000005/images/00000000-0000-0000-0000-000000000003
target_file = 00000000-0000-0000-0000-000000000004
target_format = raw
target_estimated_size = 123456789
target_overlay = /var/tmp/v2vovl344e53.qcow2
    (0/100%)    (1/100%)    (2/100%)    (3/100%)    (4/100%)    (5/100%)    (6/100%)    (7/100%)    (8/100%)    (9/100%)    (10/100%)    (11/100%)    (12/100%)    (13/100%)    (14/100%)    (15/100%)    (16/100%)    (17/100%)    (18/100%)    (19/100%)    (20/100%)    (21/100%)    (22/100%)    (23/100%)    (24/100%)    (25/100%)    (26/100%)    (27/100%)    (28/100%)    (29/100%)    (30/100%)    (31/100%)    (32/100%)    (33/100%)    (34/100%)    (35/100%)    (36/100%)    (37/100%)    (38/100%)    (39/100%)    (40/100%)    (41/100%)    (42/100%)    (43/100%)    (44/100%)    (45/100%)    (46/100%)    (47/100%)    (48/100%)    (49/100%)    (50/100%)    (51/100%)    (52/100%)    (53/100%)    (54/100%)    (55/100%)    (56/100%)    (57/100%)    (58/100%)    (59/100%)    (60/100%)    (61/100%)    (62/100%)    (63/100%)    (64/100%)    (65/100%)    (66/100%)    (67/100%)    (68/100%)    (69/100%)    (70/100%)    (71/100%)    (72/100%)    (73/100%)    (74/100%)    (75/100%)    (76/100%)    (77/100%)    (78/100%)    (79/100%)    (80/100%)    (81/100%)    (82/100%)    (83/100%)    (84/100%)    (85/100%)    (86/100%)    (87/100%)    (88/100%)    (89/100%)    (90/100%)    (91/100%)    (92/100%)    (93/100%)    (94/100%)    (95/100%)    (96/100%)    (97/100%)    (98/100%)    (99/100%)    (100/100%)[ 204.0] Creating output metadata
[ 204.0] Finishing off
